app/views/step_by_step_pages/schedule.html.erb

```diff
@@ -34,59 +34,77 @@
 <div class="row">
   <div class="col-md-7">
     <%= form_tag do %>
-      <%= render "govuk_publishing_components/components/date_input", {
-        legend_text: legend,
-        name: "schedule[date]",
-        hint: "For example, 01 08 2027",
-        id: "date",
-        error_items: issues_for(:date),
-        items: [
-          {
-            name: "day",
-            width: 2,
-            value: @schedule_placeholder[:day]
+      <% unless @change_note %>
+        <%= render "step_by_step_pages/change_notes" %>
+
+        <div class="govuk-grid-row">
+          <div class="govuk-grid-column-full">
+            <%= render "govuk_publishing_components/components/button", {
+              text: "Continue"
+            } %>
+          </div>
+          <div class="govuk-grid-column-full govuk-!-margin-top-3">
+            <%= link_to 'Cancel', @step_by_step_page, class: "govuk-link" %>
+          </div>
+        </div>
+      <% else %>
+        <%= render "govuk_publishing_components/components/date_input", {
+          legend_text: legend,
+          name: "schedule[date]",
+          hint: "For example, 01 08 2027",
+          id: "date",
+          error_items: issues_for(:date),
+          items: [
+            {
+              name: "day",
+              width: 2,
+              value: @schedule_placeholder[:day]
+            },
+            {
+              name: "month",
+              width: 2,
+              value: @schedule_placeholder[:month]
+            },
+            {
+              name: "year",
+              width: 4,
+              value: @schedule_placeholder[:year]
+            }
+          ]
+        } %>
+
+        <%= render "components/autocomplete", {
+          id: "time",
+          name: "schedule[time]",
+          label: {
+            text: "Time",
+            bold: true,
           },
-          {
-            name: "month",
-            width: 2,
-            value: @schedule_placeholder[:month]
+          input: {
+            options: time_options,
+            value: @schedule_placeholder[:time],
           },
-          {
-            name: "year",
-            width: 4,
-            value: @schedule_placeholder[:year]
-          }
-        ]
-      } %>
+          error_items: issues_for(:time),
+          data_attributes: {
+            "autocomplete-without-narrowing-results": true,
+          },
+          width: "narrow",
+        } %>
 
-      <%= render "components/autocomplete", {
-        id: "time",
-        name: "schedule[time]",
-        label: {
-          text: "Time",
-          bold: true,
-        },
-        input: {
-          options: time_options,
-          value: @schedule_placeholder[:time],
-        },
-        error_items: issues_for(:time),
-        data_attributes: {
-          "autocomplete-without-narrowing-results": true,
-        },
-        width: "narrow",
-      } %>
+        <%= hidden_field :schedule, :update_type, :value => @change_note[:update_type] %>
```

Comment:
I struggled to think of another way of persisting the `update_type` and `change_note` values through the form without writing them to the database or using sessions.  I didn't want to write anything to the database at this point, since it wouldn't be useful until the final step is completed and the step-by-step is actually scheduled.  Each POST request is entirely independent of the other, so I'm not sure there's another way of matching up the first and second requests.

Reply:
So `@change_note` is lost when the second form is submitted?

Reply:
Yes.  There's no way for the application to know the second request relates to the first request.

Reply:
I wonder if `@change_note` can be passed a param to the schedule action as suggested here: https://stackoverflow.com/questions/40227933/rails-how-to-pass-value-from-one-action-to-another-in-the-same-controller

Reply:
I've managed to solve this using `session`